state/providers.go

```diff
@@ -0,0 +1,33 @@
+package state
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/ethereum/go-ethereum/ethclient"
+)
+
+func CheckWeb3Providers(providersURIs []string) (map[int64]string, error) {
+	if len(providersURIs) == 0 {
+		return nil, fmt.Errorf("no URIs provided")
+	}
+
+	providers := make(map[int64]string)
+	for _, uri := range providersURIs {
+		cli, err := ethclient.Dial(uri)
+		if err != nil {
+			return nil, fmt.Errorf("error dialing web3 provider uri '%s': %w", uri, err)
+		}
+
+		ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
```

Comment:
Ideally these context timeout should be declared as constants.

Reply:
Yep, I have a task to review all the contexts timeouts